sentry-apollo/src/main/java/io/sentry/apollo/SentryApolloInterceptor.kt

```diff
@@ -0,0 +1,105 @@
+package io.sentry.apollo
+
+import com.apollographql.apollo.api.Mutation
+import com.apollographql.apollo.api.Query
+import com.apollographql.apollo.api.Subscription
+import com.apollographql.apollo.exception.ApolloException
+import com.apollographql.apollo.interceptor.ApolloInterceptor
+import com.apollographql.apollo.interceptor.ApolloInterceptor.CallBack
+import com.apollographql.apollo.interceptor.ApolloInterceptor.FetchSourceType
+import com.apollographql.apollo.interceptor.ApolloInterceptor.InterceptorRequest
+import com.apollographql.apollo.interceptor.ApolloInterceptor.InterceptorResponse
+import com.apollographql.apollo.interceptor.ApolloInterceptorChain
+import io.sentry.HubAdapter
+import io.sentry.IHub
+import io.sentry.ISpan
+import io.sentry.SpanStatus
+import java.util.concurrent.Executor
+
+class SentryApolloInterceptor(
+    private val hub: IHub = HubAdapter.getInstance(),
+    private val beforeSpan: BeforeSpanCallback? = null
+) : ApolloInterceptor {
+
+    constructor(hub: IHub) : this(hub, null)
+    constructor(beforeSpan: BeforeSpanCallback) : this(HubAdapter.getInstance(), beforeSpan)
+
+    override fun interceptAsync(request: InterceptorRequest, chain: ApolloInterceptorChain, dispatcher: Executor, callBack: CallBack) {
+        val activeSpan = hub.span
+        if (activeSpan == null) {
+            chain.proceedAsync(request, dispatcher, callBack)
+        } else {
+            val span = startChild(request, activeSpan)
+            val sentryTraceHeader = span.toSentryTrace()
+
+            // we have no access to URI, no way to verify tracing origins
+            val headers = request.requestHeaders.toBuilder().addHeader(sentryTraceHeader.name, sentryTraceHeader.value).build()
+            val requestWithHeader = request.toBuilder().requestHeaders(headers).build()
+            span.setTag("operationId", requestWithHeader.operation.operationId())
+            span.setTag("variables", requestWithHeader.operation.variables().valueMap().toString())
```

Comment:
a map as a tag isn't a good idea, whats about adding it as `data`? but then we are blocked by https://github.com/getsentry/sentry-java/issues/1233

Reply:
the reason is that this tag can be very long, so it's not useful to be searchable and it'll be truncated by the UI anyway

Reply:
Agree. We have to be careful with what tags we're adding out of the box as this results in potential costs on the server when high cardinality tags are used

Please default to `Contexts` or even `Extra` and if it should be a tag, lets discuss internally to consider a protocol change before adding as a tag

Reply:
span does not have access to `contexts` or `extras` but `data`, so we have to work on https://github.com/getsentry/sentry-java/issues/1233 before going forward.

@maciejwalkowiak how `variables` are important? is that the map that tells the user which query has been used or what? if its something we can add later, lets do it later, if this is the most important part to identify where the span comes from, then we have to work on #1233

Reply:
`data` is the way to go

Reply:
@marandaneto variables add more context to the request, but its not absolutely neccessary. Removed tags, lets add variables to `data` later once #1233 is there.